File: server/app/models/menu_item.py
from sqlalchemy import Column, String, Integer, Boolean, ForeignKey, Index, DateTime
from sqlalchemy.dialects.postgresql import UUID
from datetime import datetime
from uuid import uuid4
from app.db.base import Base


class MenuItem(Base):
    __tablename__ = "menu_items"
    
    id = Column(UUID(as_uuid=True), primary_key=True, default=uuid4)
    seller_id = Column(UUID(as_uuid=True), ForeignKey("users.id", ondelete="CASCADE"))
    name = Column(String, nullable=False)
    photo_url = Column(String)
    category = Column(String)
    price_pk = Column(Integer, nullable=False)
    portion = Column(String)
    is_available = Column(Boolean, default=True)
    created_at = Column(DateTime, default=datetime.utcnow)
    
    __table_args__ = (
        Index("idx_menu_items_seller", "seller_id", "is_available"),
    )

File: app/lib/data/models/menu_item.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'menu_item.freezed.dart';
part 'menu_item.g.dart';

@freezed
class MenuItem with _$MenuItem {
  const factory MenuItem({
    required String id,
    @JsonKey(name: 'seller_id') required String sellerId,
    required String name,
    @JsonKey(name: 'photo_url') String? photoUrl,
    String? category,
    @JsonKey(name: 'price_pk') required int pricePk,
    String? portion,
    @JsonKey(name: 'is_available') @Default(true) bool isAvailable,
    @JsonKey(name: 'created_at') required DateTime createdAt,
  }) = _MenuItem;

  factory MenuItem.fromJson(Map<String, dynamic> json) => _$MenuItemFromJson(json);
}

@freezed
class MenuItemCreate with _$MenuItemCreate {
  const factory MenuItemCreate({
    required String name,
    String? photoUrl,
    String? category,
    required int pricePk,
    String? portion,
    @Default(true) bool isAvailable,
  }) = _MenuItemCreate;

  factory MenuItemCreate.fromJson(Map<String, dynamic> json) =>
      _$MenuItemCreateFromJson(json);
}

